### ISSUE TITLE
Syncing New Pages Can Error with "409 Conflict"

### ISSUE THREAD

**Issue by byronmccollum** [kovetskiy/mark#139]
Possibly a race condition when creating new pages. There seems to be two distinct operations, as indicated by the page's history:

1. Create New Blank Page
2. Sync Page Contents

We are seeing the following error when creating new pages:

```
FATAL Confluence API returned unexpected status: 409 Conflict, output: ""
```

It is believed the above two phases happening in quick succession might be the cause. A resync will allow the second operation to succeed.

**Comment by byronmccollum** [comment]
We added some retry logic to our CI, and as you can see in the logs, this might be related to page locking.

```
> Syncing team/retry-test.md
272021-11-19 18:59:05.137 INFO  processing team/retry-test.md
282021-11-19 18:59:05.578 INFO  page will be stored under path: GCT Home > Team > Retry Test
292021-11-19 18:59:05.999 INFO  the leading H1 heading will be excluded from the Confluence output
302021-11-19 18:59:06.160 FATAL Confluence API returned unexpected status: 409 Conflict, output: ""
31Retry 1/5 exited 1, retrying in 1 seconds...
322021-11-19 18:59:07.176 INFO  processing team/retry-test.md
332021-11-19 18:59:07.552 INFO  page will be stored under path: GCT Home > Team > Retry Test
342021-11-19 18:59:07.603 INFO  the leading H1 heading will be excluded from the Confluence output
352021-11-19 18:59:08.009 INFO  edit locked on page "Retry Test" by user "[MASKED]" to prevent manual edits
362021-11-19 18:59:08.131 INFO  page successfully updated: [MASKED]/spaces/GCT/pages/2630353989/Retry+Test
37[MASKED]/spaces/GCT/pages/2630353989/Retry+Test
```

**Comment by rajpalparyani** [comment]
I got the same error. How did you add the retry logic? Can you provide an example?

**Comment by JAndritsch** [comment]
I'm seeing the same thing. I ran `mark` to create several new pages. The page that threw an exception was created but has no content. Here is the error returned by Confluence during the "Update" call for that page:

2022-04-11 13:38:08.997 FATAL Confluence API returned unexpected status: 409 Conflict, output: "{\"statusCode\":409,\"data\":{\"authorized\":false,\"valid\":true,\"errors\":[],\"successful\":false},\"message\":\"com.atlassian.confluence.api.service.exceptions.ConflictException: javax.persistence.OptimisticLockException: Row was updated or deleted by another transaction (or unsaved-value mapping was incorrect) : [com.atlassian.confluence.pages.Page#3279323546]\"}"
